packages/wrangler/src/__tests__/traverse-module-graph.test.ts

```diff
@@ -0,0 +1,181 @@
+			"./other.js",
+			dedent/* javascript */ `
+			export const HELLO = "WORLD"
+			`
+		);
+
+		const bundle = await traverseModuleGraph(
+			{
+				file: path.join(process.cwd(), "./index.js"),
+				directory: process.cwd(),
+				format: "modules",
+				moduleRoot: process.cwd(),
+			},
+			[{ type: type as ConfigModuleRuleType, globs: ["**/*.js"] }]
+		);
+
+		expect(bundle.modules[0].type).toStrictEqual(format);
+	});
+	it("should not resolve JS without module root", async () => {
+		await mkdir("./src/nested", { recursive: true });
+		await writeFile(
+			"./src/nested/index.js",
+			dedent/* javascript */ `
+			import { HELLO } from "../other.js"
+			export default {
+				async fetch(request) {
+					return new Response(HELLO)
+				}
+			}
+			`
+		);
+		await writeFile(
+			"./src/other.js",
+			dedent/* javascript */ `
+			export const HELLO = "WORLD"
+			`
+		);
+
+		const bundle = await traverseModuleGraph(
+			{
+				file: path.join(process.cwd(), "./src/nested/index.js"),
+				directory: path.join(process.cwd(), "./src/nested"),
+				format: "modules",
+				// The default module root is dirname(file)
+				moduleRoot: path.join(process.cwd(), "./src/nested"),
+			},
+			[{ type: "ESModule", globs: ["**/*.js"] }]
+		);
+
+		expect(bundle.modules).toStrictEqual([]);
+	});
+	it("should resolve JS with module root", async () => {
+		await mkdir("./src/nested", { recursive: true });
+		await writeFile(
+			"./src/nested/index.js",
+			dedent/* javascript */ `
+			import { HELLO } from "../other.js"
+			export default {
+				async fetch(request) {
+					return new Response(HELLO)
+				}
+			}
+			`
+		);
+		await writeFile(
+			"./src/other.js",
+			dedent/* javascript */ `
+			export const HELLO = "WORLD"
+			`
+		);
+
+		const bundle = await traverseModuleGraph(
+			{
+				file: path.join(process.cwd(), "./src/nested/index.js"),
+				directory: path.join(process.cwd(), "./src/nested"),
+				format: "modules",
+				// The default module root is dirname(file)
+				moduleRoot: path.join(process.cwd(), "./src"),
+			},
+			[{ type: "ESModule", globs: ["**/*.js"] }]
+		);
+
+		expect(bundle.modules[0].name).toStrictEqual("other.js");
+	});
+	it("should resolve non-JS without rules", async () => {
+		await mkdir("./src", { recursive: true });
+		await writeFile(
+			"./src/index.js",
+			dedent/* javascript */ `
+			import HELLO from "../other.txt"
+			export default {
+				async fetch(request) {
+					return new Response(HELLO)
+				}
+			}
+			`
+		);
+		await writeFile(
+			"./src/other.txt",
+			dedent/* javascript */ `
+			export const HELLO = "WORLD"
+			`
+		);
+
+		const bundle = await traverseModuleGraph(
+			{
+				file: path.join(process.cwd(), "./src/index.js"),
+				directory: path.join(process.cwd(), "./src"),
+				format: "modules",
+				// The default module root is dirname(file)
+				moduleRoot: path.join(process.cwd(), "./src"),
+			},
+			[]
+		);
+
+		expect(bundle.modules[0].name).toStrictEqual("other.txt");
+	});
+});
```

Comment:
Is it worth adding a test where the glob does not match the file even though it is within the module root?

Reply:
[c0bc8f3](https://github.com/cloudflare/workers-sdk/pull/2769/commits/c0bc8f39e30baa2e7551e42b8274269c722b8e44)